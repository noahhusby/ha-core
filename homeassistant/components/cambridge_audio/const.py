"""Constants for the Cambridge Audio integration."""

import asyncio
import logging

from aiostreammagic import StreamMagicConnectionError, StreamMagicError

DOMAIN = "cambridge_audio"

SERVICE_PLAY_AIRABLE = "play_airable"
ATTR_AIRABLE_ID = "airable_radio_id"
ATTR_AIRABLE_NAME = "airable_radio_name"

SERVICE_PLAY_RADIO = "play_radio"
ATTR_RADIO_NAME = "radio_name"
ATTR_RADIO_URL = "radio_url"
LOGGER = logging.getLogger(__package__)

STREAM_MAGIC_EXCEPTIONS = (
    StreamMagicConnectionError,
    StreamMagicError,
    asyncio.CancelledError,
    TimeoutError,
)

CONNECT_TIMEOUT = 5
