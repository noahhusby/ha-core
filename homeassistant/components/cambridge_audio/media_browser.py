"""Support for media browsing."""

from homeassistant.components.media_player import BrowseMedia, MediaClass
from homeassistant.core import HomeAssistant


async def async_browse_media(
    hass: HomeAssistant,
    media_content_id: str | None,
    media_content_type: str | None,
) -> BrowseMedia:
    """Browse media."""
    if media_content_id is None:
        return await root_payload(
            hass,
        )


async def root_payload(
    hass: HomeAssistant,
) -> BrowseMedia:
    return BrowseMedia(
        title="Cambridge Audio",
        media_class=MediaClass.DIRECTORY,
        media_content_id="",
        media_content_type="root",
        can_play=False,
        can_expand=True,
    )
