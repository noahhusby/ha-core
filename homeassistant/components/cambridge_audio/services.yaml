play_airable:
  target:
    device:
      integration: cambridge_audio
  fields:
    airable_radio_id:
      required: true
      selector:
        number:
          min: 0
          max: 9999
          mode: box
    station_name:
      default: "Radio"
      selector:
        text:

play_radio:
  target:
    device:
      integration: cambridge_audio
  fields:
    radio_url:
      required: true
      selector:
        text:
    station_name:
      default: "Radio"
      selector:
        text:
