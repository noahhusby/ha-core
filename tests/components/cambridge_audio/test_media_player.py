"""Tests for the Cambridge Audio media player."""
from unittest.mock import AsyncMock, patch

import pytest
from syrupy import SnapshotAssertion
from syrupy.filters import props

from homeassistant.components.cambridge_audio.media_player import CambridgeAudioDevice
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_registry import RegistryEntry
from homeassistant.util.decorator import Registry
from tests.common import MockConfigEntry, snapshot_platform
from tests.components.cambridge_audio import setup_integration
from homeassistant.helpers import entity_registry as er

async def setup(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_stream_magic_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> RegistryEntry:
    """Test all entities."""
    with patch("homeassistant.components.cambridge_audio.PLATFORMS", [Platform.MEDIA_PLAYER]):
        await setup_integration(hass, mock_config_entry)

    entity_entries = er.async_entries_for_config_entry(entity_registry, mock_config_entry.entry_id)
    return entity_entries[0]


