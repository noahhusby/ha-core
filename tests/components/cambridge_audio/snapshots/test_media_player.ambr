# serializer version: 1
# name: test_all_entities[media_player.cambridge_audio_cxnv2-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'source_list': list([
        'Internet Radio',
        'USB Audio',
        'D2',
        'D1',
        'Media Library',
        'AirPlay',
        'Spotify',
        'Chromecast built-in',
        'Roon Ready',
        'TIDAL Connect',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'media_player',
    'entity_category': None,
    'entity_id': 'media_player.cambridge_audio_cxnv2',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'cambridge_audio',
    'previous_unique_id': None,
    'supported_features': <MediaPlayerEntityFeature: 18865>,
    'translation_key': None,
    'unique_id': '0020c2d8',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[media_player.cambridge_audio_cxnv2-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'entity_picture': '/api/media_player_proxy/media_player.cambridge_audio_cxnv2',
      'friendly_name': 'Cambridge Audio CXNv2',
      'media_album_name': "Greatest Hits: God's Favorite Band",
      'media_artist': 'Green Day',
      'media_content_type': <MediaType.MUSIC: 'music'>,
      'media_duration': 232,
      'media_position': 179,
      'media_position_updated_at': 0,
      'media_title': 'Holiday',
      'source': 'AirPlay',
      'source_list': list([
        'Internet Radio',
        'USB Audio',
        'D2',
        'D1',
        'Media Library',
        'AirPlay',
        'Spotify',
        'Chromecast built-in',
        'Roon Ready',
        'TIDAL Connect',
      ]),
      'supported_features': <MediaPlayerEntityFeature: 18865>,
    }),
    'context': <ANY>,
    'entity_id': 'media_player.cambridge_audio_cxnv2',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'playing',
  })
# ---
